src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -174,6 +174,29 @@ namespace QgsWms
     return image.release();
   }
 
+  QJsonObject *QgsRenderer::getLegendGraphicsAsJson( QgsLayerTreeModel &model )
+  {
+    // get layers
+    std::unique_ptr<QgsLayerRestorer> restorer;
+    restorer.reset( new QgsLayerRestorer( mContext.layers() ) );
+
+    // configure layers
+    QList<QgsMapLayer *> layers = mContext.layersToRender();
+    configureLayers( layers );
+
+    // init renderer
+    QgsLegendSettings settings = legendSettings();
+    QgsLegendRenderer renderer( &model, settings );
+
+    // rendering
+    std::unique_ptr<QJsonObject> json;
+    json.reset( new QJsonObject() );
+    QgsRenderContext renderContext;
+    renderer.exportLegendToJson( renderContext, *json );
```

Comment:
json.get() ?

Reply:
`json.get()` returns a pointer to the managed object. What I need to pass to exportLegendToJson is a reference to the object, and `*` dereferences the pointer to the managed object. But I am not a C++ expert :-)

Reply:
`*(json.get())` would probably work as well, but `*json` does the same as far as I understand.

Reply:
Yes, it does.
But I don't get the whole use of pointers here, why do you need a unique ptr? Or a pointer at all.
You could return by value and rely on RVO.

Reply:
Yes, you are certainly right. I'll look into it.